src/webviews/extension-side/dataviewer/kernelDataViewerDependencyImplementation.ts

```diff
@@ -0,0 +1,121 @@
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { executeSilently } from '../../../kernels/helpers';
+import { IKernel } from '../../../kernels/types';
+import { parseSemVer } from '../../../platform/common/utils';
+import { IDataViewerDependencyService } from './types';
+
+export const minimumSupportedPandaVersion = '0.20.0';
+export const kernelGetPandasVersion =
+    'import pandas as _VSCODE_pandas;print(_VSCODE_pandas.__version__);del _VSCODE_pandas';
+
+function isVersionOfPandaSupported(version: SemVer) {
+    return version.compare(minimumSupportedPandaVersion) > 0;
+}
+
+function kernelPackaging(kernel: IKernel): '%conda' | '%pip' {
+    const envType = kernel.kernelConnectionMetadata.interpreter?.envType;
+    const isConda = envType === EnvironmentType.Conda;
+    console.log({ envType, isConda });
+    // From https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-pip (%conda is here as well).
+    return isConda ? '%conda' : '%pip';
+}
+
+/**
+ * Uses the Kernel to manage the dependencies of a Data Viewer.
+ */
+export class KernelDataViewerDependencyImplementation implements IDataViewerDependencyService {
+    constructor(private readonly applicationShell: IApplicationShell, private isCodeSpace: boolean) {}
+
+    protected async execute(command: string, kernel: IKernel): Promise<(string | undefined)[]> {
+        const outputs = await executeSilently(kernel.session!, command);
+        const error = outputs.find((item) => item.output_type === 'error');
+        if (error) {
+            traceWarning(DataScience.failedToGetVersionOfPandas(), error.message);
+        }
+        return outputs.map((item) => item.text?.toString());
+    }
+
+    protected async getVersion(kernel: IKernel): Promise<SemVer | undefined> {
+        try {
+            const outputs = await this.execute(kernelGetPandasVersion, kernel);
+            return outputs.map((text) => (text ? parseSemVer(text.toString()) : undefined)).find((item) => item);
+        } catch (e) {
+            traceWarning(DataScience.failedToGetVersionOfPandas(), e.message);
+            return;
+        }
+    }
+
+    private async installMissingDependencies(kernel: IKernel): Promise<void> {
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!
+        });
+
+        let selection: string | undefined;
+        if (kernel) {
+            selection = this.isCodeSpace
+                ? Common.install()
+                : await this.applicationShell.showErrorMessage(
+                      DataScience.pandasRequiredForViewing(),
+                      { modal: true },
+                      Common.install()
+                  );
+        } else {
+            await this.applicationShell.showErrorMessage(DataScience.pandasRequiredForViewing());
+        }
+
+        if (!kernel) {
+            return;
```

Comment:
not necessary, don't see how this is possible

Reply:
This is what happens when I take a PR out of draft a Friday night 😅 thank you! I'm reviewing my PR. Thank you for your patience 🙏